addons/mrp/static/src/js/mrp.js

```diff
@@ -122,7 +122,17 @@ var TimeCounter = AbstractField.extend({
         } else {
             clearTimeout(this.timer);
         }
-        this.$el.html($('<span>' + moment.utc(this.duration).format("HH:mm:ss") + '</span>'));
+
+        function padStart(number){
+            return number.toString().padStart(2, '0')
+        }
+
+        const duration = moment.duration(this.duration, 'milliseconds');
+        const hours = Math.floor(duration.asHours());
+        const minutes = Math.floor(duration.asMinutes()) - hours * 60;
+        const seconds = Math.floor(duration.asSeconds() - (hours * 3600) - (minutes * 60));
```

Comment:
It looks like all this hard work can just be replaced by `duration.minutes()` and `duration.seconds()` which resolve the corresponding field exactly.